enable SDHC support on imx93 Cortex-A Core platform

JiafeiPan opened the pull request:
> imx usdhc driver updates: 

1. clang format the driver
2. include soc.h conditionally
3. enable MMIP mapping

> Enable USDHC support for imx93 Cortex-A Core platform

Depend on https://github.com/zephyrproject-rtos/hal_nxp/pull/524, it has been merged.

JiafeiPan committed:
drivers: sdhc: imx_usdhc: enable MMIO mapping

Enable MMIO mapping in the driver.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
drivers: sdhc: imx_usdhc: run clang-format on the driver

Format the driver by using clang-format.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
dts: arm64: imx93: add USDHC device nodes

Add device nodes for SDHC.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: imx93_evk: enable SDHC on the board

Added pinctrl and dts nodes for uSDHC1 and uSDHC2, they are disabled
by default, and provided overlay files to enable them if needed.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>
Signed-off-by: Lei Xu <lei.xu@nxp.com>

danieldegrasse commented on boards/nxp/imx93_evk/doc/index.rst:
```diff
@@ -108,6 +110,23 @@ Serial Port
 This board configuration uses a single serial communication channel with the
 CPU's UART2 for A55 core and M33 core.
 
+uSDHC (SD or eMMC Interface)
```
Assuming this can't be enabled on the board by default (is this USDHC interface shared with Linux/Uboot?)- can we add some form of testcase definition in `tests/subsys/sd/sdmmc` to validate that the subsystem works on this board?

JiafeiPan replied:
Hi, @danieldegrasse, good catch,  currently we rely on uboot or linux to start A-core zephyr, so Zephyr can't use the same uSDHC controller with uboot or Linux, that is to say:
1) if board is booted from SD Card (which uses uSDHC2), so we only could use uSDHC1 for Zephyr `tests/subsys/sd/mmc` testing.
2) if board is booted from mmc (which uses uSDHC1), so we only could use uSDHC2 for Zephyr `tests/subsys/sd/sdmmc` testing

I will update this document and provides overlay for both uSDHC controllers to address this issues, thanks.

JiafeiPan commented:
V2: Addressed @danieldegrasse's comments, added two overlay for each uSDHC  controller, and updated document to address need to avoid conflict with U-Boot and Linux.

JiafeiPan commented:
Hi, @danieldegrasse , would you please review this PR again? thanks.

JiafeiPan commented:
Rebased.

ithinuel commented on drivers/sdhc/imx_usdhc.c:
```diff
@@ -13,7 +13,9 @@
 #include <zephyr/drivers/clock_control.h>
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/logging/log.h>
+#if __has_include("soc.h")
 #include <soc.h>
+#endif
```
`soc.h` is also part of the [soc porting guide](https://docs.zephyrproject.org/latest/hardware/porting/soc_porting.html#create-your-soc-directory).

JiafeiPan replied:
Created an empty soc.h, thanks.

JiafeiPan committed:
soc: imx93: a55: add empty soc.h

Some drivers need header file soc.h, according to Zephyr SoC Porting Guide
soc.h must be provided for each SoC, so created an empty one.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan commented:
Added one empty soc.h to address the comments from @decsny and @ithinuel.
Rebased.

JiafeiPan commented:
The twister issue is caused by https://github.com/zephyrproject-rtos/zephyr/issues/89142

JiafeiPan commented:
> The twister issue is caused by #89142

Rebased as the issue has been fixed.

JiafeiPan commented:
Rebased and removed west.yaml update as current commit of hal_nxp hal already included the update which this PR depends on.
